packages/core/src/MatchMaker.ts

```diff
@@ -421,15 +424,15 @@ export async function gracefullyShutdown(): Promise<any> {
   }
 
   // remove processId from room count key
-  presence.hdel(getRoomCountKey(), processId);
+  await presence.hdel(getRoomCountKey(), processId);
```

Comment:
gracefullyShutdown is called as a result of unhandled exceptions. As soon as you await when processing an unhandled exception, the application will exit, thus, this will leave redis full and won't even call `disconnect`. This would be a pretty big problem when we want a chance to save data during unhandled exception.